client/trino-client/src/main/java/io/trino/client/auth/external/ExternalAuthentication.java

```diff
@@ -58,7 +58,7 @@ public ExternalAuthentication(URI tokenUri, Optional<URI> redirectUri)
                 currentUri = result.getNextTokenUri();
                 continue;
             }
-
+            poller.tokenReceived(currentUri);
```

Comment:
Can we do it async? It is not required to authenticate, but it will make authentication process longer.

Reply:
Let's leave it as it is. If we have any communication issues then doing this async won't help much.